tests_page_object_pattern/rozetka_search_results_page.py

```diff
@@ -0,0 +1,28 @@
+from selene import browser, by, be, have
+
+from basket_modal import RozetkaBasketModal
+from product_details_page import ProductDetailsPage
+
+class RozetkaSearchResultsPage:
+
+    def verify_search_results_at_least(self, number_of_results):
+        browser.all('//a/span[contains(text(), "")]').should(have.size_greater_than_or_equal(number_of_results))
+
+    def get_product_price(self, product_number):
+        return browser.element(by.xpath(format("(//span[@class='goods-tile__price-value'])[{0}]"
+                                               .format(str(product_number))))).text
+
+    def get_product_name(self, product_number):
+        return browser.element(by.xpath(format("(//span[@class='goods-tile__title'])[{0}]"
+                                               .format(str(product_number))))).text
+
+    def add_product_to_basket(self, product_number):
```

Comment:
all action methods should return self of new object

Reply:
search_result_page.add_product_to_basket(4) -> no return
search_result_page.do_smht()

search_result_page.add_product_to_basket(4).do_somth() -> return self